docs/user_guide/mitigation.rst

```diff
@@ -74,6 +74,62 @@ Preprocessing
    
 .. currentmodule:: fairlearn.preprocessing
 
+Preprocessing algortihms transform the dataset to mitigate possible unfairness
+present in the data.
+Preprocessing algorithms in Fairlearn follow the `scikit-learn transformers
+<https://scikit-learn.org/stable/modules/generated/sklearn.base.TransformerMixin.html#sklearn.base.TransformerMixin>`_
+class, meaning that they can :code:`fit` to the dataset and :code:`transform` it.
+A :code:`fit_transform` might not always be available, in which case the separate
+:code:`fit` and :code:`transform` methods need to be used in sequence.
+
+.. _correlation_remover:
+
+Correlation Remover
+~~~~~~~~~~~~~~~~~~~
+Sensitive features can be correlated with non sensitive features in the dataset.
+By applying the :code:`CorrelationRemover`, these correlations are filtered out while
+as much as possible information is retained (as measured by the least-squares error).
+This is mathemetically described as:
+
+.. math::
+        \min _{\mathbf{z}_{1}, \ldots, \mathbf{z}_{n}} \sum_{i=1}^{n}\left\|\mathbf{z}_{i}
+        -\mathbf{x}_{i}\right\|^{2} \\
+        \text{subject to} \\
+        \frac{1}{n} \sum_{i=1}^{n} \mathbf{z}_{i}\left(\mathbf{s}_{i}-\overline{\mathbf{s}}
+        \right)^{T}=\mathbf{0}
+
+In the example below, a subset of the `adult dataset <https://www.openml.org/d/1590>`_
+is loaded and the correlation between sex and the non-sensitive features is removed.
+
+.. doctest:: mitigation
+    :options:  +NORMALIZE_WHITESPACE
+
+    >>> from fairlearn.preprocessing import CorrelationRemover
+    >>> import pandas as pd
+    >>> from sklearn.datasets import fetch_openml
+    >>> data = fetch_openml(data_id=1590, as_frame=True)
+    >>> X = data.data[['age', 'fnlwgt', 'education-num', 'sex']]
+    >>> X = pd.get_dummies(X)
+    >>> cr = CorrelationRemover(sensitive_feature_ids=['sex_Female', 'sex_Male'])
+    >>> cr.fit(X)
+    CorrelationRemover(sensitive_feature_ids=['sex_Female', 'sex_Male'])
+    >>> X_transform = cr.transform(X)
+
+..  I am not sure whether it would be valuable to show the transformed data here,
```

Comment:
What do you have in mind in terms of showing?

Reply:
I had in mind to just display the transformed values, such that the user gets an i dea of what it would look like, e.g. the below screenshot. However, maybe it is not really valuable to show that to the user, hence my comment in the code :).
![Transformed data example](https://user-images.githubusercontent.com/33757199/156995915-bfed736c-fe31-4bbe-9629-dd78f397b36a.png)

Reply:
How about the following? I recall from some papers and/or lecetures that t-SNE is quite nice for these kind of visualizations, and I think it looks quite good and intuitive. I also tried PCA, but that didn't look quite as nice as the t-SNE plots.
![CR](https://user-images.githubusercontent.com/33757199/160357446-88a06edf-c94f-40c7-8ffc-cffd9d846390.png)
![No CR](https://user-images.githubusercontent.com/33757199/160357448-1f5e4893-1c78-4100-a9cb-537e47095b1d.png)

Reply:
I tried both PCA and t-SNE, but both of them showed very little difference in the before and after plots. I figured it would perhaps be most intuitive to show the correlation matrices before and after running the `CorrelationRemover`, so that is what I have done right now by making an example notebook and including the figures in the user guide :)

Reply:
Ah yes t-SNE is what I was thinking if without remembering the name, thanks! Hmm, that makes it tricky...

I agree with your process and conclusion. If any other @fairlearn/fairlearn-maintainers or others have thoughts on what might be useful to add please chime in! I guess the most important part is always how it changes outcomes, but it's really interesting how it removes correlations, too, hence a correlation matrix makes sense. 👍🏻 Might be worth pointing out the important entries, though, because people (or is that just me) don't read matrices 😄

Reply:
My suggestion would be to show a heatmap (with colors as well as numbers) of the *differences* in correlations before/after (e.g., a value of 0.2 would mean that correlation between X1 and X2 has increased with 0.2). That allows you to see which correlations were affected the most (and in what way) in a single glance.